python/cudf/cudf/core/dataframe.py

```diff
@@ -1881,23 +1881,45 @@ def pop(self, item):
         del self[item]
         return popped
 
-    def rename(self, mapper=None, columns=None, copy=True, inplace=False):
-        """
-        Alter column labels.
+    def rename(
+        self,
+        mapper=None,
+        columns=None,
+        index=None,
+        copy=True,
+        inplace=False,
+        axis=0,
+    ):
+        """Alter column and index labels.
 
         Function / dict values must be unique (1-to-1). Labels not contained in
         a dict / Series will be left as-is. Extra labels listed don’t throw an
         error.
 
+        DataFrame.rename supports two calling conventions
+
+        `(index=index_mapper, columns=columns_mapper, ...)`
+        `(mapper, axis={0/'index' or 1/'column'}, ...)`
+
+        We highly recommend using keyword arguments to clarify your intent.
+
         Parameters
         ----------
-        mapper, columns : dict-like or function, optional
-            dict-like or functions transformations to apply to
-            the column axis' values.
+        mapper : dict-like or function, optional dict-like or functions
+            transformations to apply to the index/column values depending
+            on selected `axis`.
+        columns : dict-like or function, optional dict-like or functions
+            transformations to apply to the columns axis' values.
+        index : dict-like or function, optional dict-like or functions
+            transformations to apply to the index axis' values.
```

Comment:
First line of parameters should just be types and optionally the default value.

Also, should note that we don't support functions for `index` or `axis=0` yet.